src/main.c

```diff
@@ -335,53 +365,58 @@ int main(int argc, char* argv[]) {
         }
         dumpopts |= JV_PRINT_INDENT_FLAGS(indent);
         i++;
-        if (!short_opts) continue;
+        continue;
       }
       if (isoption(argv[i], 0, "seq", &short_opts)) {
         options |= SEQ;
-        if (!short_opts) continue;
+        continue;
       }
       if (isoption(argv[i], 0, "stream", &short_opts)) {
         parser_flags |= JV_PARSE_STREAMING;
-        if (!short_opts) continue;
+        continue;
       }
       if (isoption(argv[i], 0, "stream-errors", &short_opts)) {
         parser_flags |= JV_PARSE_STREAM_ERRORS;
-        if (!short_opts) continue;
+        continue;
       }
       if (isoption(argv[i], 'e', "exit-status", &short_opts)) {
         options |= EXIT_STATUS;
         if (!short_opts) continue;
       }
       // FIXME: For --arg* we should check that the varname is acceptable
+      if (isoption(argv[i], 0, "args", &short_opts)) {
+        further_args_are_strings = 1;
+        continue;
+      }
+      if (isoption(argv[i], 0, "jsonargs", &short_opts)) {
+        further_args_are_json = 1;
```

Comment:
And here we should reset `further_args_are_strings`.  The idea is that one could intermingle `--args` and `--jsonargs`.